includes/class-newspack-newsletters.php

```diff
@@ -706,22 +707,40 @@ public static function api_set_mailchimp_interest( $request ) {
 				);
 			}
 
-			$segment_opts = ( 'no_interests' === $request['interest_id'] ) ?
-				(object) [] :
-				[
+			$has_interest  = $interest_id_param && ! $is_unsetting_interest;
+			$tag_ids_param = $request['tag_ids'];
+
+			$segment_opts = (object) [];
```

Comment:
Why the cast?

Reply:
To be explicit that `$segment_opts` is an object, not an array – it was the default before (see [removed line 710 above](https://github.com/Automattic/newspack-newsletters/pull/241/files#diff-76aefac9a481217d439a23e638846effL710))